pybossa/sched.py

```diff
@@ -255,19 +255,9 @@ def template_get_locked_task(project_id, user_id=None, user_ip=None,
             "Project {} - number of current users: {}"
             .format(project_id, user_count))
 
-        sql = query_factory(
-            project_id,
-            user_id=user_id,
-            user_ip=user_ip,
-            external_uid=external_uid,
-            limit=limit,
-            offset=offset,
-            orderby=orderby,
-            desc=desc,
-            rand_within_priority=rand_within_priority,
-            present_gold_task=present_gold_task,
-            gold_only=gold_only
-        )
+        sql = query_factory(project_id, user_id=user_id, limit=limit,
```

Comment:
Just to confirm, was it intended to remove the additional parameters to the query? For example, `user_ip`, `external_uid`, `offset`, `orderby`, etc? It looks like these have been moved into `locked_task_sql()` where the query is now dynamically assembled.

Reply:
It was intentional to remove them because they weren't used in the original `query_factory` functions.  They were noise that was just being passed along for the ride.  I removed them which should hopefully be simpler and cause less confusion.

If in the future, we want to "lock" some other `query_factory` function that needs those other params, we can update the code here.